zerver/models/users.py

```diff
@@ -1205,3 +1205,21 @@ def get_bot_dicts_in_realm(realm: "Realm") -> list[dict[str, Any]]:
 
 def is_cross_realm_bot_email(email: str) -> bool:
     return email.lower() in settings.CROSS_REALM_BOT_EMAILS
+
+
+class ExternalAuthID(models.Model):
+    user_profile = models.ForeignKey(UserProfile, on_delete=CASCADE)
+    realm = models.ForeignKey("zerver.Realm", on_delete=CASCADE)
+    date_created = models.DateTimeField(default=timezone_now)
+    # TODO: Do we want is_active and date_deactivated fields?
```

Comment:
I don't see a reason to want those until we have a UI design that would do something with the. This TODO comment can probably stay though.